starknet/starknet-stream/src/data/model.ts

```diff
@@ -0,0 +1,61 @@
+import type * as data from '@subsquid/starknet-normalization'
+import type {GetFields, Select, Selector, Simplify, TrueFields} from './util'
+import type {
+    BlockRequiredFields,
+    EventRequiredFields,
+    TransactionRequiredFields
+} from './data-partial'
+
+export type FELT = string
+
+
+export interface FieldSelection {
```

Comment:
currenlty FieldSelection allows to specify fields that aren't available in the gateway. i tried to request all fields and got this error message:
{"body":{"title":"400 Bad Request","description":"{'fields': {'block': defaultdict(<class 'dict'>, {'l1GasPriceInFri': {'key': ['Invalid value.']}, 'l1GasPriceInWei': {'key': ['Invalid value.']}, 'starknetVersion': {'key': ['Invalid value.']}}), 'transaction': defaultdict(<class 'dict'>, {'blockNumber': {'key': ['Invalid value.']}}), 'event': defaultdict(<class 'dict'>, {'blockNumber': {'key': ['Invalid value.']}, 'key0': {'key': ['Invalid value.']}, 'key1': {'key': ['Invalid value.']}, 'key2': {'key': ['Invalid value.']}, 'key3': {'key': ['Invalid value.']}, 'keysSize': {'key': ['Invalid value.']}, 'restKeys': {'key': ['Invalid value.']}})}}"}}

field selection should match to the one that is defined in archive.py
https://github.com/subsquid/archive.py/blob/master/sqa/starknet/query.py#L43

Reply:
Thanks a lot, I have initially built data models according to tables in archive.py and havent rechecked since. For events I will need one more workaround because required fields fundamentally different from actual structure. For current implementation this means optional keys will always be presented in object received in handler.
https://github.com/Wizard1209/squid-sdk/blob/starknet-support/starknet/starknet-stream/src/data/model.ts#L52

Reply:
i guess we can keep keys as an array. squid developers will have to deal with indexes on their own.
my proposal is to keep EventRequest as it is. but starknet-normalization shouldn't split keys into separate fields and Event entity should contain only keys field.
that's what we do with log topics in evm